neuro_san/internals/run_context/langchain/core/langchain_run_context.py

```diff
@@ -585,21 +586,27 @@ async def wait_on_run(self, run: Run, journal: Journal = None) -> Run:
             # to the logs.  Add this because some people are interested in it.
             callbacks.append(LoggingCallbackHandler(self.logger))
 
+        # Set up a run name for tracing purposes
+        metadata: Dict[str, Any] = self.invocation_context.get_metadata()
+        run_name: str = metadata.get("request_id", "<unknown>") + "-" + \
+            Origination.get_full_name_from_origin(self.origin)
+
         # Add callbacks as an invoke config
         invoke_config = {
             "configurable": {
                 "session_id": run.get_id()
             },
             "callbacks": callbacks,
-            "recursion_limit": recursion_limit
+            "recursion_limit": recursion_limit,
+            "run_name": run_name
```

Comment:
Add this notion of run_name to the invoke_config. It ends up making the LangSmith display much more intelligible.